ocaml/xapi/storage_impl.ml

```diff
@@ -692,8 +692,9 @@ module Wrapper = functor(Impl: Server_impl) -> struct
 
     let destroy context ~dbg ~dp ~allow_leak =
       info "DP.destroy dbg:%s dp:%s allow_leak:%b" dbg dp allow_leak;
-      let failures = List.map (fun (sr, sr_t) -> destroy_sr context ~dbg ~dp ~sr ~sr_t ~allow_leak false) (Host.list !Host.host)
-                   |>List.filter(function | None -> false | Some x -> true)
+      let failures = Host.list !Host.host
+		   |>List.map (fun (sr, sr_t) -> destroy_sr context ~dbg ~dp ~sr ~sr_t ~allow_leak false)
+                   |>List.filter((<>) None)
                    |>List.map(function | None -> assert false | Some x -> x) in
```

Comment:
Thanks - will change. I came across the Core implementation, should've checked if you had one too